tests/test_utils/logging_setup.py

```diff
@@ -54,4 +54,10 @@ def setup_logging() -> None:
     log_level = os.environ.get("SYNAPSE_TEST_LOG_LEVEL", "ERROR")
     root_logger.setLevel(log_level)
 
+    # Log when events are (maybe unexpectedly) filtered out of responses in tests. It's
+    # just nice to be able to look at the CI log and figure out why an event isn't being
+    # returned.
+    logging.getLogger("synapse.visibility.filtered_event_debug").setLevel(logging.DEBUG)
```

Comment:
I don't feel strongly, but i find it a little odd to enable this for default in tests when we seem to go to great lengths to not log anything?

Reply:
> we seem to go to great lengths to not log anything?

Is this based off the fact that we only log at the `ERROR` level by default when running trial tests?

hmm, I am used to always using `SYNAPSE_TEST_LOG_LEVEL=INFO` in my trial test runs so I didn't consider this. Although we do use `INFO` level for Complement test runs by default.

Perhaps based on this, we don't want it by default for trial tests but would be fine for Complement. I was originally was wanting this while investigating things in a Complement context anyway.

Reply:
> > we seem to go to great lengths to not log anything?
> 
> Is this based off the fact that we only log at the `ERROR` level by default when running trial tests?

Mostly, yes.

Reply:
Maybe turn it on automatically if the logger is set to INFO or DEBUG?